### PR TITLE
Extend and enhance README.md #37

### LINKED ISSUE #37: Client-side code generation
**Is your feature request related to a problem? Please describe.**
Currently, the main purpose of the plugin is to generate server-side code (POJOs and interfaces). 
The idea is to add support of client-side classes generation. This will allow users to build a GraphQL request.

**Describe the solution you'd like**
Schema example:
```graphql
type Mutation {
    newBike(bikeInput: BikeInput!): Bike
}

input BikeInput {
    type: BikeType!
    brand: String!
    size: String!
    price: BigDecimal
}

type Bike {
    id: ID
    type: BikeType
    brand: String
    size: String
    price: BigDecimal
    addedDateTime: DateTime
}
```

Generated class:

```java
public class NewBikeMutationRequest {
    private BikeInput bikeInput;
    //getters/setters
}

public class BikeInputTO {
    private BikeTypeTO type;
    private String brand;
    private String size;
    private String price;

    public BikeInputTO() { }

    //getters/setters
```

The end-user will then use these classes to build the request and **response projection** (open question for now). So that it will be serialized to the following request string:

```graphql
mutation newBike {
  newBike(bike: {
    type: ROAD,
    brand: "LOL",
    size: "Big",
    year: 2019,
    price: "123.45"
  }) {
    id
    addedDateTime
  }
}
```

### PR DESCRIPTION
**Pull request by kobylynskyi**
Closes #37

### CODE REVIEW

File: plugins/gradle/README.md [hunk 1]
```diff
@@ -11,9 +11,12 @@ This document describes the gradle plugin for graphql-java-codegen.
 This Gradle plugin is able to generate the following classes based on your GraphQL schema:
 * Interfaces for GraphQL queries, mutations and subscriptions
 * Interfaces for GraphQL unions
-* POJO classes for GraphQL types
-* Enum classes for each GraphQL enum
-* Interface Resolvers for GraphQL type fields 
+* POJO classes for GraphQL types/inputs
+* Enum classes for GraphQL enums
+* Interface Resolvers for GraphQL type fields
```
Comment (joffrey-bion) [inline plugins/gradle/README.md:16]: I don't quite get this one, is it the same as the last?

File: plugins/gradle/README.md [hunk 2]
```diff
@@ -165,8 +175,8 @@ Later on you can call each task separately or together:
 
 ### Convert generated Java classes to Kotlin classes
 
-Navigate in IntelijIdea to the `./build/generated/graphql/` folder and press `Cmd+Alt+Shift+K`
-Access to classes from your code as normal Kotlin classes.
+1. Navigate in InteliJ IDEA to the `./build/generated/graphql/` folder and press `Cmd+Alt+Shift+K`
```
Comment (joffrey-bion) [inline plugins/gradle/README.md:178]: Typo InteliJ -> IntelliJ

Comment (kobylynskyi) [reply plugins/gradle/README.md]: Thanks! Fixed.

File: plugins/gradle/README.md (hunk 1 above)
Comment (kobylynskyi) [reply plugins/gradle/README.md:16]: You are totally right.
I removed the last statement.

**Comment by kobylynskyi** [comment]
@joffrey-bion I've added option `generateDataFetchingEnvironmentArgumentInApis` (from #66) to all README files.

File: plugins/maven/README.md [hunk 3]
```diff
@@ -56,31 +56,33 @@ This Maven plugin is able to generate the following classes based on your GraphQ
...
+| `modelPackage`                                  | String             | Empty                                     | Java package for generated model classes (type, input, interface, enum, union). |
+| `generateBuilder`                               | Boolean            | True                                      | Specifies whether generated model classes should have builder. |
+| `generateApis`                                  | Boolean            | True                                      | Specifies whether api classes should be generated as well as model classes. |
+| `generateAsyncApi`                              | Boolean            | False                                     | If true, then wrap type into `java.util.concurrent.CompletableFuture` or `subscriptionReturnType` |
+| `generateDataFetchingEnvironmentArgumentInApis` | Boolean            | False                                     | If true, then `graphql.schema.DataFetchingEnvironment env` will be added as a last argument to all methods of root type resolvers and field resolvers. |
```
Comment (joffrey-bion) [inline plugins/maven/README.md:71]: Sorry for coming up with such a long name, I tried removing some of the words but it really feels like they are all needed 😅

Comment (kobylynskyi) [reply plugins/maven/README.md:71]: Haha, true 🤣 
Spring-style!

**Pull request merged**